src/features/defi/providers/foxy/components/FoxyManager/Deposit/FoxyDeposit.tsx

```diff
@@ -0,0 +1,661 @@
+import { TransactionReceipt } from 'web3-core/types'
+import { Amount } from 'components/Amount/Amount'
+import { CircularProgress } from 'components/CircularProgress/CircularProgress'
+import { MiddleEllipsis } from 'components/MiddleEllipsis/MiddleEllipsis'
+import { RouteSteps, StatusTextEnum } from 'components/RouteSteps/RouteSteps'
+import { Row } from 'components/Row/Row'
+import { Text } from 'components/Text'
+import { useBrowserRouter } from 'context/BrowserRouterProvider/BrowserRouterProvider'
+import { useChainAdapters } from 'context/PluginProvider/PluginProvider'
+import { useWallet } from 'context/WalletProvider/WalletProvider'
+import { bnOrZero } from 'lib/bignumber/bignumber'
+import { poll } from 'lib/poll/poll'
+import { marketApi } from 'state/slices/marketDataSlice/marketDataSlice'
+import {
+  selectAssetByCAIP19,
+  selectMarketDataById,
+  selectPortfolioCryptoBalanceByAssetId,
+  selectPortfolioLoading
+} from 'state/slices/selectors'
+import { useAppDispatch, useAppSelector } from 'state/store'
+
+import { FoxyDepositActionType, initialState, reducer } from './DepositReducer'
+
+enum DepositPath {
+  Deposit = '/',
+  Approve = '/approve',
+  ApproveSettings = '/approve/settings',
+  Confirm = '/confirm',
+  ConfirmSettings = '/confirm/settings',
+  Status = '/status'
+}
+
+export const routes = [
+  { step: 0, path: DepositPath.Deposit, label: 'Deposit' },
+  { step: 1, path: DepositPath.Approve, label: 'Approve' },
+  { path: DepositPath.ApproveSettings, label: 'Approve Settings' },
+  { step: 2, path: DepositPath.Confirm, label: 'Confirm' },
+  { path: DepositPath.ConfirmSettings, label: 'Confirm Settings' },
+  { step: 3, path: DepositPath.Status, label: 'Status' }
+]
+
+export type FoxyDepositProps = {
+  api: FoxyApi
+}
+
+export const FoxyDeposit = ({ api }: FoxyDepositProps) => {
+  const [state, dispatch] = useReducer(reducer, initialState)
+  const location = useLocation()
+  const history = useHistory()
+  const appDispatch = useAppDispatch()
+  const translate = useTranslate()
+  const { query, history: browserHistory } = useBrowserRouter<DefiQueryParams, DefiParams>()
+  const { chain, contractAddress, tokenId } = query
+  const alertText = useColorModeValue('blue.800', 'white')
+  const defaultStatusBg = useColorModeValue('white', 'gray.700')
+  const network = NetworkTypes.MAINNET
+  const assetNamespace = AssetNamespace.ERC20
+  const assetCAIP19 = caip19.toCAIP19({ chain, network, assetNamespace, assetReference: tokenId })
+  const feeAssetCAIP19 = caip19.toCAIP19({
+    chain,
+    network,
+    assetNamespace: AssetNamespace.Slip44,
+    assetReference: AssetReference.Ethereum
+  })
+
+  const asset = useAppSelector(state => selectAssetByCAIP19(state, assetCAIP19))
+  const marketData = useAppSelector(state => selectMarketDataById(state, assetCAIP19))
+  if (!marketData) appDispatch(marketApi.endpoints.findByCaip19.initiate(assetCAIP19))
```

Comment:
const { loading, data, error } = useFindByCaip19()

stylistic but also gives you a loading state if you want